packages/ui/src/forms/RenderFields/RenderField.tsx

```diff
@@ -85,6 +85,18 @@ export const RenderField: React.FC<Props> = ({
       />
     )
   } else {
+    if ('fields' in fieldComponentProps.field && Array.isArray(fieldComponentProps.field.fields)) {
```

Comment:
This loop is very expensive. It will run on every render for every field. If you need to add a property onto the field at runtime, do it in the client config.